docs/content/Deployment/Platform-Kubernetes.md

```diff
@@ -0,0 +1,516 @@
+permalink: /deployment/platforms/kubernetes
+category: Deployment
+subCategory: Platforms
+menuOrder: 1
+---
+
+This guide walks you through deploying Cube.js with Kubernetes. This particular
+deployment makes use of a `hostPath` volume to mount schema files into the
+containers.
+
+<!-- prettier-ignore-start -->
+[[warning |]]
+| This is an example of a production-ready deployment, but real-world
+| deployments can vary significantly depending on desired performance
+| and scale.
+<!-- prettier-ignore-end -->
+
+## Prerequisites
+
+- [Kubernetes][k8s]
+
+## Configuration
+
+### Create Cube.js API instance and Refresh Worker
+
+We'll start by creating a Redis deployment in a file called
+`redis-deployment.yaml`:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    io.kompose.service: redis
+  name: redis
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      io.kompose.service: redis
+  strategy: {}
+  template:
+    metadata:
+      labels:
+        io.kompose.service: redis
+    spec:
+      containers:
+        - env:
+            - name: ALLOW_EMPTY_PASSWORD
+              value: 'yes'
+          image: bitnami/redis:latest
+          name: redis
+          ports:
+            - containerPort: 6379
+          resources: {}
+      restartPolicy: Always
+status: {}
+```
+
+And a corresponding service for the Redis deployment in a file named
+`redis-service.yaml`:
+
+```yaml
+apiVersion: v1
+kind: Service
+metadata:
+  labels:
+    io.kompose.service: redis
+  name: redis
+spec:
+  ports:
+    - name: '6379'
+      port: 6379
+      targetPort: 6379
+  selector:
+    io.kompose.service: redis
+status:
+  loadBalancer: {}
+```
+
+Next, create a file called `cube-api-deployment.yaml` with the following
+contents:
+
+```yaml
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    io.kompose.service: cube-api
+  name: cube-api
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      io.kompose.service: cube-api
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        io.kompose.service: cube-api
+    spec:
+      containers:
+        - env:
+            - name: CUBEJS_API_SECRET
+              value: secret
+            - name: CUBEJS_CUBESTORE_HOST
+              value: 'cubestore-router'
+            - name: CUBEJS_CUBESTORE_PORT
+              value: '3030'
+            - name: CUBEJS_DB_CLICKHOUSE_READONLY
+              value: 'true'
+            - name: CUBEJS_DB_HOST
+              value: github.demo.trial.altinity.cloud
+            - name: CUBEJS_DB_NAME
+              value: default
+            - name: CUBEJS_DB_PASS
+              value: demo
+            - name: CUBEJS_DB_PORT
+              value: '8443'
+            - name: CUBEJS_DB_SSL
+              value: 'true'
+            - name: CUBEJS_DB_TYPE
+              value: clickhouse
+            - name: CUBEJS_DB_USER
+              value: demo
+            - name: CUBEJS_DEV_MODE
+              value: 'true'
```

Comment:
Set to `false` and add a comment stating not to run `CUBEJS_DEV_MODE: true` in prod.